hedera-node/src/main/java/com/hedera/services/store/contracts/precompile/SyntheticTxnFactory.java

```diff
@@ -430,11 +432,39 @@ public TransactionBody.Builder createAccount(final Key alias, final long balance
     }
 
     public TransactionBody.Builder nodeStakeUpdate(
-            final Timestamp stakingPeriodEnd, final List<NodeStake> nodeStakes) {
+            final Timestamp stakingPeriodEnd,
+            final List<NodeStake> nodeStakes,
+            final PropertySource properties) {
+        final var stakingRewardRate = dynamicProperties.getStakingRewardRate();
+        final var threshold = dynamicProperties.getStakingStartThreshold();
+        final var stakingPeriod = properties.getLongProperty("staking.periodMins");
+        final var stakingPeriodsStored =
+                properties.getIntProperty("staking.rewardHistory.numStoredPeriods");
```

Comment:
Define these strings as constants somewhere, preferably in `PropertySource`.

Or alternatively, why not just add concrete getters to it like `propertySource.getStakingNumStoredPeriods()` or `propertySource.getStaking().getRewardHistory().getNumStoredPeriods()`?

Reply:
Yes agreed. There are a lot of properties [here](https://github.com/hashgraph/hedera-services/blob/master/hedera-node/src/main/java/com/hedera/services/context/properties/BootstrapProperties.java), that need to be changed as well. Will open a different PR defining all the Strings as `CONSTANTS` and will replace the used places.

Reply:
Opened this [issue](https://github.com/hashgraph/hedera-services/issues/3799)